drivers/video/gc2145.c

```diff
@@ -0,0 +1,1150 @@
+	{0x4e, 0x07},
+	{0x4c, 0x02},
+	{0x4d, 0xc8},
+	{0x4e, 0x07},
+	{0x4c, 0x02},
+	{0x4d, 0xe8},
+	{0x4e, 0x07},
+	{0x4c, 0x02},
+	{0x4d, 0xa7},
+	{0x4e, 0x07},
+	{0x4c, 0x02},
+	{0x4d, 0xc7},
+	{0x4e, 0x07},
+	{0x4c, 0x02},
+	{0x4d, 0xe7},
+	{0x4e, 0x07},
+	{0x4c, 0x03},
+	{0x4d, 0x07},
+	{0x4e, 0x07},
+
+	{0x4f, 0x01},
+	{0x50, 0x80},
+	{0x51, 0xa8},
+	{0x52, 0x47},
+	{0x53, 0x38},
+	{0x54, 0xc7},
+	{0x56, 0x0e},
+	{0x58, 0x08},
+	{0x5b, 0x00},
+	{0x5c, 0x74},
+	{0x5d, 0x8b},
+	{0x61, 0xdb},
+	{0x62, 0xb8},
+	{0x63, 0x86},
+	{0x64, 0xc0},
+	{0x65, 0x04},
+	{0x67, 0xa8},
+	{0x68, 0xb0},
+	{0x69, 0x00},
+	{0x6a, 0xa8},
+	{0x6b, 0xb0},
+	{0x6c, 0xaf},
+	{0x6d, 0x8b},
+	{0x6e, 0x50},
+	{0x6f, 0x18},
+	{0x73, 0xf0},
+	{0x70, 0x0d},
+	{0x71, 0x60},
+	{0x72, 0x80},
+	{0x74, 0x01},
+	{0x75, 0x01},
+	{0x7f, 0x0c},
+	{0x76, 0x70},
+	{0x77, 0x58},
+	{0x78, 0xa0},
+	{0x79, 0x5e},
+	{0x7a, 0x54},
+	{0x7b, 0x58},
+	{0xfe, 0x00},
+	{0xfe, 0x02},
+	{0xc0, 0x01},
+	{0xc1, 0x44},
+	{0xc2, 0xfd},
+	{0xc3, 0x04},
+	{0xc4, 0xF0},
+	{0xc5, 0x48},
+	{0xc6, 0xfd},
+	{0xc7, 0x46},
+	{0xc8, 0xfd},
+	{0xc9, 0x02},
+	{0xca, 0xe0},
+	{0xcb, 0x45},
+	{0xcc, 0xec},
+	{0xcd, 0x48},
+	{0xce, 0xf0},
+	{0xcf, 0xf0},
+	{0xe3, 0x0c},
+	{0xe4, 0x4b},
+	{0xe5, 0xe0},
+	{0xfe, 0x01},
+	{0x9f, 0x40},
+	{0xfe, 0x00},
+
+	/* Output Control  */
+	{0xfe, 0x00},
+	{0xf2, 0x0f},
+	{0xfe, 0x02},
+	{0x40, 0xbf},
+	{0x46, 0xcf},
+	{0xfe, 0x00},
+
+	{0xfe, 0x00},
+	{0x05, 0x01},
+	{0x06, 0x1C},
+	{0x07, 0x00},
+	{0x08, 0x32},
+	{0x11, 0x00},
+	{0x12, 0x1D},
+	{0x13, 0x00},
+	{0x14, 0x00},
+
+	{0xfe, 0x01},
+	{0x3c, 0x00},
+	{0x3d, 0x04},
+	{0xfe, 0x00},
+	{0x00, 0x00},
+};
+
+struct gc2145_config {
+	struct i2c_dt_spec i2c;
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	struct gpio_dt_spec reset_gpio;
+#endif
+};
+
+struct gc2145_data {
+	struct video_format fmt;
+};
+
+#define GC2145_VIDEO_FORMAT_CAP(width, height, format) \
+	{                                              \
+	.pixelformat = format,                         \
+	.width_min = width,                            \
+	.width_max = width,                            \
+	.height_min = height,                          \
+	.height_max = height,                          \
+	.width_step = 0,                               \
+	.height_step = 0,                              \
+	}
+
+static const struct video_format_cap fmts[] = {
+	GC2145_VIDEO_FORMAT_CAP(1600, 1200, VIDEO_PIX_FMT_RGB565), /* UXGA  */
+	{0},
+};
+
+static int gc2145_write_reg(const struct i2c_dt_spec *spec, uint8_t reg_addr, uint8_t value)
+{
+	int ret;
+	uint8_t tries = 3;
+
+	/*
+	 * It rarely happens that the camera does not respond with ACK signal.
+	 * In that case it usually responds on 2nd try but there is a 3rd one
+	 * just to be sure that the connection error is not caused by driver
+	 * itself.
+	 */
+	do {
+		ret = i2c_reg_write_byte_dt(spec, reg_addr, value);
+		if (!ret) {
+			return 0;
+		}
+		/* If writing failed wait 5ms before next attempt */
+		k_msleep(5);
```

Comment:
Pointing at it in case you wanted to convert all occurrences of `k_msleep()` to `k_sleep(K_MSEC())`...